scl/sumologic/sumologic.conf

```diff
@@ -0,0 +1,82 @@
+#############################################################################
+# Copyright (c) 2020 Balabit
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+# EXAMPLES:
+#
+# With TLS encryption (make sure you trust the sumologic CA cert by putting it
+# to /etc/ssl, or create a separate CA directory):
+# For endpoint see - https://help.sumologic.com/APIs/General-API-Information/Sumo-Logic-Endpoints-and-Firewall-Security
+# 
+# Send data using syslog:
+#    log {
+#       destination {
+#                sumologic-syslog(token("USER-TOKEN-AS-PROVIDED-BY-sumologic") deployment("ENDPOINT")
+#                       tls(peer-verify(required-trusted) ca-dir('/etc/syslog-ng/ca.d'))
+#                );
+#       };
+#    };
+#
+#
+# Send JSON data:
+#    log {
+#        source { system(); };
+#        destination {
+#            sumologic-syslog(token("USER-TOKEN-AS-PROVIDED-BY-sumologic") deployment("ENDPOINT")
+#                   tls(peer-verify(required-trusted) ca-dir('/etc/syslog-ng/ca.d')) template("$(format-json --scope all-nv-pairs)"));
+#            );
+#        };
+#    };
+#
+# Send data using http:
+#    log {
+#        destination {
+#            sumologic-http(collector("UNIQUE-HTTP-COLLECTOR-CODE-AS-PROVIDED-BY-sumologic") deployment("ENDPOINT")
+#                   tls(peer-verify(yes) ca-dir('/etc/syslog-ng/ca.d')));
+#            );
+#        };
+#    };
+#
+
+@requires json-plugin
```

Comment:
This is fine, but recently there has been a huge improvement on the `@requires` usage in this PR https://github.com/syslog-ng/syslog-ng/pull/3015.

The current version, if you do not have the `json-plugin` it is going to skip both `sumologic-syslog` and `sumologic-http` despite the later could work without json just fine (as far as I can see the http version does not require json plugin).

While if you move the `@requires json-plugin` inside the `block` and add an extra error message after it, it is going to provide nicer error message (see the linekd PR) and also only exclude that `block` and not the rest of the file.

My suggestion is somthing like this:
```
block destination sumologic-syslog(... )
{
@requires json-plugn "The json-plugin is required for sumologic-syslog(), ..."
...
}
```

Reply:
@Kokan : okay, thanks for your review.